blueprints/materials/concrete.py

```diff
@@ -1 +1,547 @@
+        e_c: MPA
+            Use a custom modulus of elasticity of concrete [MPa]
+            If none is given the value from table 3.1 (Ecm) of NEN-EN 1992-1-1 is used.
+        cement_class: CementClass
+            Enumeration of cement class (default= N)
+        density: KG
+            Density in [kg/m³] (default= 2500.0)
+        diagram_type: DiagramType
+            Enumeration of diagram types of stress-strain relations (default= Bi-Linear)
+        aggregate_type: ConcreteAggregateType
+            Enumeration of concrete aggregate types (default= Quartzite)
+        aggregate_size: MM
+            Largest nominal maximum aggregate size [mm] (NEN-EN 1992-1-1) (default= 16.0)
+        plain_concrete_diagram: bool
+            Use a Stress-strain diagram for plain or lightly reinforced concrete (NEN-EN 1992-1-1) (default= False)
+        material_factor: DIMENSIONLESS
+            Partial safety factor (yc) for concrete according to NEN-EN 1992-1-1 art.2.4.2.4 (default= 1.5)
+        thermal_coefficient: DIMENSIONLESS
+            Thermal coefficient of the material [-] (default= 1e-5)
+        cement_type: CementType
+            Cement types (default= CEM I)
+
+        """
+        self._e_c = e_c
+        self._cement_class: CementClass = cement_class
+        self._concrete_class: ConcreteStrengthClass = concrete_class
+        self._name: str = name
+        self._density: float = density
+        self._diagram_type: DiagramType = diagram_type
+        self._aggregate_type: ConcreteAggregateType = aggregate_type
+        self._aggregate_size: float = aggregate_size
+        self._plain_concrete_diagram: bool = plain_concrete_diagram
+        self._material_factor: float = material_factor
+        self._thermal_coefficient = thermal_coefficient
+        self._cement_type: CementType = cement_type
+
+    @property
+    def name(self) -> str:
+        """Name of the concrete material.
+
+        Returns
+        -------
+        str
+            Example: "C30/37"
+        """
+        if self._name:
+            return self._name
+        return self._concrete_class.value
+
+    @property
+    def e_c(self) -> MPA:
+        """Modulus of elasticity of concrete [MPa].
+
+        If no custom value is given, the value from table 3.1 of NEN-EN 1992-1-1 is used.
+
+        Returns
+        -------
+        MPA
+            Example: 32836 (for C30/37)
+        """
+        if self._e_c:
+            return self._e_c
+        return self.e_cm
+
+    @property
+    def concrete_class(self) -> ConcreteStrengthClass:
+        """Concrete class of the ConcreteMaterial object.
+
+        Returns
+        -------
+        ConcreteStrengthClass
+            Example: "ConcreteStrengthClass.C30_37"
+        """
+        return self._concrete_class
+
+    @property
+    def material_factor(self) -> DIMENSIONLESS:
+        """Partial safety factor (yc) for concrete according to NEN-EN 1992-1-1 art.2.4.2.4[-].
```

Comment:
in general: I would not include the unit in the 'summary' line. add it to the returns chapter where it belongs.